Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs

```diff
@@ -364,28 +365,22 @@ public IEnumerable<UserAc> GetAllUsers()
         /// </summary>
         /// <param name="email"></param>
         /// <returns> boolean: true if the email exists, false if does not exist</returns>
-        public async Task<bool> FindByEmail(string email)
+        public async Task<bool> CheckEmailIsExists(string email)
         {
             var user = await _userManager.FindByEmailAsync(email);
             if (user == null)
-            {
                 return false;
-            }
-            return true;
+            else
+                return true;
         }
 
-
-        public bool FindUserBySlackUserName(string slackUserName)
+        public ApplicationUser FindUserBySlackUserName(string slackUserName)
         {
             var user = _applicationUserDataRepository.FirstOrDefault(x => x.SlackUserName == slackUserName);
-            if (user != null)
-            {
-                if (user.SlackUserName == slackUserName)
-                {
-                    return false;
-                }
-            }
-            return true;
+            if (user == null)
+                throw(new SlackUserNotFound());
```

Comment:
You don't need to add brackets in throwing an exception. Simply write `throw new SlackUserNotFound()`